tabular/src/autogluon/tabular/learner/abstract_learner.py

```diff
@@ -138,6 +138,21 @@ def predict(self, X: DataFrame, model=None, as_pandas=False):
     def _validate_fit_input(self, X: DataFrame, **kwargs):
         if self.label not in X.columns:
             raise KeyError(f"Label column '{self.label}' is missing from training data. Training data columns: {list(X.columns)}")
+        self.weight_column = kwargs.get('weight_column', None)
```

Comment:
I think it might be cleaner to just have `sample_weight` be the name since it is widely used in other packages. We can simply explain that it must be a str column name via ValueError:

```
if sample_weight is not None and not isinstance(sample_weight, str):
    raise ValueError(f'sample_weight must be of string type indicating the column name of sample weight in the data. Instead, sample_weight was of type: {type(sample_weight)}')
```